pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -681,13 +682,13 @@ private void doAuthentication(PGStream pgStream, String host, String user, Prope
                 LOGGER.log(Level.FINEST, "<=BE AuthenticationReqPassword");
                 LOGGER.log(Level.FINEST, " FE=> Password(password=<not shown>)");
 
-                byte[] encodedPassword = AuthenticationPluginManager.getEncodedPassword(AuthenticationRequestType.CLEARTEXT_PASSWORD, info);
-
-                pgStream.sendChar('p');
-                pgStream.sendInteger4(4 + encodedPassword.length + 1);
-                pgStream.send(encodedPassword);
-                pgStream.sendChar(0);
-                pgStream.flush();
+                AuthenticationPluginManager.withEncodedPassword(AuthenticationRequestType.CLEARTEXT_PASSWORD, info, encodedPassword -> {
+                  pgStream.sendChar('p');
+                  pgStream.sendInteger4(4 + encodedPassword.length + 1);
+                  pgStream.send(encodedPassword);
+                  pgStream.sendChar(0);
+                  pgStream.flush();
```

Comment:
I guess those sentChar and flush could be moved out of `with...Password`

Reply:
Yes I was of two minds on that. I originally wanted to pull them out to make the time it's in memory as minimal as possible. But then the code seemed harder to follow.

Reply:
Security requirements to make code harder to follow.

For instance, I guess we need to reset `pgStream` buffers after `flush` or something like that.

Reply:
Maybe but there's no way I'm touching in this PR. That'd definitely have wider consequences and warrant wider testing.